tests/nested/core/core20-tpm/task.yaml

```
@@ -79,7 +79,7 @@ execute: |
     tests.nested exec "sudo reboot" || true
     tests.nested wait-for reboot "${boot_id}"
     echo "Check the test will race with snapd updating the modeenv, wait for the modenv to have a good state"
-    nested_retry_until_success 10 5 "grep -E \"current_kernel_command_lines=.*snapd_recovery_mode=run\" < /var/lib/snapd/modeenv"
+    tests.nested exec "retry --wait 5 -n 10 sh -c 'MATCH current_kernel_command_lines=.*snapd_recovery_mode=run < /var/lib/snapd/modeenv'"
```

```suggestion
    retry --wait 5 -n 10 sh -c 'tests.nested exec cat /var/lib/snapd/modeenv' | MATCH current_kernel_command_lines=.*snapd_recovery_mode=run'
```

this one too I think

---

I'm not sure we can drop the quoting here and in the other places if we keep tests.nested exec. I'm guessing there will be a problem passing quoted arguments to the underlying retry tool once again.. So in the end is a question of whether it's more convenient to have `retry .. 'tests.nested exec ..'` or `tests.neste exec 'retry ..'`

---

as I explained in my review, regardless of quoting I think it is more readable to have the SSH connection re-made every time to run the command, this may produce extra login messages in the URL, but it does mean that if the SSH connection goes away, we will have better output as the SSH connection will fail to be made and presumably we will see that quicker from trying to make a new connection than having an existing connection just break randomly which IME seems to take a bit longer to propagate back to the output